core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java

```diff
@@ -882,6 +882,11 @@ private <K, V> CompletionStage<MarshallableEntry<K, V>> loadFromStoresIterator(O
          CompletionStage<MarshallableEntry<K, V>> loadStage = store.load(segmentOrZero(storeStatus, segment), key);
          return loadStage.thenCompose(e -> {
             if (e != null) {
+               // Read only we apply lifespan expiration to the entry, so it can be reread later
+               if (storeStatus.hasCharacteristic(Characteristic.READ_ONLY) && configuration.expiration().lifespan() > 0) {
+                  e = marshallableEntryFactory.cloneWithExpiration((MarshallableEntry) e, timeService.wallClockTime(),
+                        configuration.expiration().lifespan(), configuration.expiration().maxIdle());
```

Comment:
Looking closer maxIdle should not be supported here. We only "allow" max idle when the store has passivation, which means it can't be read only. https://github.com/infinispan/infinispan/blob/b9b2337b564564f35d280e17a12c5162ad465133/core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java#L224